FlowCal/io.py

```diff
@@ -164,6 +164,12 @@ def read_fcs_text_segment(buf, begin, end, delim=None):
     buf.seek(begin)
     raw = buf.read((end+1)-begin)
 
+    # Some files have been found to include space characters at the end of the
+    # TEXT segment. If the delimiter is not a space character, strip spaces
+    # from the end of the segment.
+    if not delim.isspace():
+        raw = raw.rstrip()
+
```

Comment:
maybe you should cut off everything past the last instance of the delimiter? this is a minor distinction, but cutting off all white space may not be the same as eliminating everything after the last delimiter. if, on the other hand, some files do not include a final delimiter at the end of the TEXT segment, this would throw away valid values, which would be bad.